cpp/Documentation.md

```diff
@@ -0,0 +1,474 @@
+# C++ Project Documentation
+
+This document provides instructions and hints to document C++ projects.
+
+## How to document code?
+
+### C++ API
+
+Public APIs may be documented with Doxygen, with comments specified in
+headers so that they remain visible to the developers consuming your APIs.
+
+### C++ Internal Code
+
+Documentation of internal code is highly recommended to help the maintainers
+of the project. Using Doxygen is not mandatory though. But if used, then the
+documentations of public and internal code should be well separated, such
+that it remains easy to the reader to distinguish public symbols from internal
+ones.
+
+Documentation of internal symbols can be specified in cpp files where symbols
+are defined, presumably more accessible to the maintainers.
+See [the following discussion](https://stackoverflow.com/questions/355619/where-to-put-the-doxygen-comment-blocks-for-an-internal-library-in-h-or-in-cpp)
+on stackoverflow.
+
+#### Documentation style
+
+Doxygen supports [multiple syntax](http://www.doxygen.nl/manual/docblocks.html)
+to document the code. There is no preferred syntax but Doxygen comments of a particular
+project may use the same one.
+
+Nevertheless, it is suggested to turn the
+[JAVADOC_AUTOBRIEF](http://www.doxygen.nl/manual/config.html#cfg_javadoc_autobrief)
+option to `YES` to shrink comments size:
+
+```c++
+/// A brief function description.
```

Comment:
does that mean that the first line is always automatically `\brief` ?
If yes, I'm in favor.

Reply:
yes!